src/mongocxx/collection.hpp

```diff
@@ -477,6 +477,16 @@ class MONGOCXX_API collection {
     stdx::string_view name() const noexcept;
 
     ///
+    /// Rename this collection.
+    ///
+    /// @param new_name The new name to assign to the collection.
+    /// @param overwrite_old Whether to overwrite any
+    ///   existing collections called new_name. Default: false
+    /// @throws exception::operation if the operation fails.
+    void name(stdx::string_view new_name, bool overwrite_old = false);
```

Comment:
sure, I will change both.
`drop_target_before_rename` hmm. IMO overwrite_old is much more clear.
`rename` Sure about that? I deliberately elected not to call it rename, because this implements a getter/setter pattern. If mongocxx generally uses getters/setters I think it would be more idiomatic to consider the name a gettable and settable property too.